pkg/clusterversion/cv.go

```diff
@@ -76,9 +80,26 @@ func (c *clusterVersionClient) EnsureDesiredConfig(uc *upgradev1alpha1.UpgradeCo
 
 	desired := uc.Spec.Desired
 
-	// Commence upgrade using version and channel
-	// Move the cluster to the same channel first
-	if !empty(desired.Channel) && !empty(desired.Version) {
+	// Check which upgrade spec source we are going to use
+	upgradeSource, err := checkUpgradeSource(uc)
+	if err != nil {
+		return false, err
+	}
+	switch upgradeSource {
+	// Use image to upgrade if the spec.desired.image is present
+	case UpgradeWithImage:
+		if clusterVersion.Spec.DesiredUpdate == nil {
+			logger.Info(fmt.Sprintf("Setting ClusterVersion to Image %s", desired.Image))
+			clusterVersion.Spec.DesiredUpdate = &configv1.Update{Image: desired.Image}
+			err = c.client.Update(context.TODO(), clusterVersion)
+			if err != nil {
+				return false, err
+			}
+			return true, nil
+		}
+		return false, fmt.Errorf("clusterversion already had desiredUpdate set, might be pervious upgrade wasn't completed successfully")
```

Comment:
This section also has an issue in that it will prevent any further image-based upgrades from occuring once the cluster has upgraded once.

When a cluster finishes upgrading, `desiredUpdate` remains in place in `clusterversion`. So once a cluster has upgraded once, this will never be `nil`. A second upgrade will cause it to drop down to this error path.

Reply:
hmm, I removed the `clusterVersion.Spec.DesiredUpdate.Image != desired.Image` from the original code, but did not test with the second upgrade. Thanks for catch this.